### PR TITLE
Partial refunds frontend

### PR DESCRIPTION
**Pull request by BilalG1**
https://www.loom.com/share/bb7abfde507f40d386ee856f5ffbd506

## Summary by CodeRabbit

* **New Features**
  * USD-based refund system enabling partial and full refunds with explicit USD amounts
  * Per-entry refund selection with granular quantity controls in refund dialogs
* **Bug Fixes**
  * Stronger refund validation and error handling to prevent invalid or out-of-bounds refunds
* **Tests**
  * Expanded end-to-end coverage for refund edge cases and scenarios
* **Style**
  * Improved refund dialog UI with contextual alerts and better controls

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/template/src/lib/stack-app/apps/interfaces/admin-app.ts [hunk 1]
```diff
@@ -109,7 +110,12 @@ export type StackAdminApp<HasTokenStore extends boolean = boolean, ProjectId ext
       { teamId: string, itemId: string, quantity: number, expiresAt?: string, description?: string } |
       { customCustomerId: string, itemId: string, quantity: number, expiresAt?: string, description?: string }
     )): Promise<void>,
-    refundTransaction(options: { type: "subscription" | "one-time-purchase", id: string }): Promise<void>,
+    refundTransaction(options: {
+      type: "subscription" | "one-time-purchase",
+      id: string,
+      amountUsd: MoneyAmount,
+      refundEntries: Array<{ entryIndex: number, quantity: number }>,
+    }): Promise<void>,
```
Comment (N2D4) [inline packages/template/src/lib/stack-app/apps/interfaces/admin-app.ts:118]: isn't amountUsd also just an entry for the refundEntries?

**Review by N2D4 — APPROVED** [review]
